feat: scanner blog post (#2422)
huggingface/blog: protectai.md

---
title: "Hugging Face Teams Up with Protect AI: Enhancing Model Security for the Community"
thumbnail: /blog/assets/protectai-partnership/thumbnail.png
authors:
- user: mcpotato
---

We are very pleased to announce our partnership with Protect AI, as part of our [long-standing commitment](https://huggingface.co/blog/2024-security-features) to provide a safe and reliable platform for the community.

[Protect AI](https://protectai.com/) is a company founded with a mission to create a safer AI powered world. They are developing powerful tools, namely [Guardian](https://protectai.com/guardian), to ensure that the rapid pace of AI innovation can continue without compromising on security.

Our decision to partner with Protect AI stems from their [community driven](https://huntr.com/) approach to security, active support of [open source](https://github.com/protectai) and expertise in all things security x AI.

## Model security refresher

> To share models, we serialize the data structures we use to interact with the models, in order to facilitate storage and transport. Some serialization formats are vulnerable to nasty exploits, such as arbitrary code execution (looking at you pickle), making sharing models potentially dangerous.

Reviewer: Love the shame directed at pickle.

> As Hugging Face has become the de facto platform for model sharing, we’d like to protect the community from this, hence why we have developed tools like [picklescan](https://github.com/mmaitre314/picklescan) and why we are integrating Guardian in our scanner suite.

Reviewer: alt: we'd like to protect our community from this, 

This way it implies ownership and engagement with the HF community overall.

> Pickle is not the only exploitable format out there, [see for reference](https://github.com/Azure/counterfit/wiki/Abusing-ML-model-file-formats-to-create-malware-on-AI-systems:-A-proof-of-concept) how one can exploit Keras Lambda layers to achieve arbitrary code execution. The good news is that Guardian catches both kinds of exploits!

Reviewer: catches both of these exploits and more!

> Pickle is not the only exploitable format out there, [see for reference](https://github.com/Azure/counterfit/wiki/Abusing-ML-model-file-formats-to-create-malware-on-AI-systems:-A-proof-of-concept) how one can exploit Keras Lambda layers to achieve arbitrary code execution. The good news is that Guardian catches both kinds of exploits!

Reviewer 2: ```suggestion
Pickle is not the only exploitable format out there, [see for reference](https://github.com/Azure/counterfit/wiki/Abusing-ML-model-file-formats-to-create-malware-on-AI-systems:-A-proof-of-concept) how one can exploit Keras Lambda layers to achieve arbitrary code execution. The good news is that Guardian catches both of these exploits and more in additional file formats! See their [Knowledge Base](https://insights-db.protectai.dev/insights/knowledge-base/) for up to date scanner information.
```

---
title: "Hugging Face Teams Up with Protect AI: Enhancing Model Security for the ML Community"

> thumbnail: /blog/assets/protectai-partnership/thumbnail.png

Reviewer 3: ```suggestion
thumbnail: /blog/assets/protectai/thumbnail.png
```

Author: ah good catch, thought I'd done it

authors:
- user: mcpotato
---

We are pleased to announce our partnership with Protect AI, as part of our [long-standing commitment](https://huggingface.co/blog/2024-security-features) to provide a safe and reliable platform for the ML community.

> [Protect AI](https://protectai.com/) is a company founded with a mission to create a safer AI powered world. They are developing powerful tools, namely [Guardian](https://protectai.com/guardian), to ensure that the rapid pace of AI innovation can continue without compromising on security.

Reviewer 3: ```suggestion
[Protect AI](https://protectai.com/) is a company founded with a mission to create a safer AI-powered world. They are developing powerful tools, namely [Guardian](https://protectai.com/guardian), to ensure that the rapid pace of AI innovation can continue without compromising security.
```

> Our decision to partner with Protect AI stems from their [community driven](https://huntr.com/) approach to security, active support of [open source](https://github.com/protectai) and expertise in all things security x AI.

Author: Sorry about the commas, I keep missing them, French be damned!

> [!TIP]
> Read all our documentation on security here: https://huggingface.co/docs/hub/security 🔥

## Integration

While integrating Guardian as a third-party scanner, we have used this as an opportunity to revamp our frontend to display scan results. Here is what it now looks like:

<img class="block" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hub/third-party-scans-list.png"/>

<img class="block" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hub/security-scanner-status-banner.png"/>

<img class="block" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hub/security-scanner-pickle-import-list.png"/>
<em>As you can see here, an additional Pickle button is present when a pickle import scan occurred</em>

As you can see from the screenshot, you have nothing to do to benefit from this! All public model repositories will be scanned by Guardian automatically as soon as you push your files to the Hub. Here is an example repository you can check out to see the feature in action: [mcpotato/42-eicar-street](https://huggingface.co/mcpotato/42-eicar-street).

> Note that you might not see a scan for your model as of today, as we have over 1 million model repos, it may take some time to catch up.

Reviewer 3: ```suggestion
Note that you might not see a scan for your model as of today, as we have over 1 million model repos. It may take us some time to catch up 😅.
```
